apostrophecms/apostrophe
# add cache invalidate field

Part of [PRO-2576](https://linear.app/apostrophecms/issue/PRO-2576/caching-emit-etags-for-the-context-document-and-invalidate-them-based)

## Summary

- Add a `cacheInvalidatedAt` field via a migration for docs that don't have it yet, and set it to equal their `updatedAt` field value
- Set `cacheInvalidatedAt` value to the current date when saving a doc

## What are the specific steps to test this change?

- Run the website

👉 `add-cache-invalidated-at-field-for-*` migration should be done only once:

```
Running database migration: add-cache-invalidated-at-field
```

👉 `cacheInvalidatedAt` field should have been added to all documents from the `aposDocs` collection, and set to equal the docs' `updatedAt` field.

- Add a piece, or a page

👉 The document created (piece or page for instance) should have a `cacheInvalidatedAt` that equals the `updatedAt` field.

## What kind of change does this PR introduce?
*(Check at least one)*


## Make sure the PR fulfills these requirements:


If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.
